engine/class_modules/sc_shaman.cpp

```diff
@@ -4374,6 +4395,11 @@ struct lava_burst_overload_t : public elemental_overload_spell_t
       m *= 1.0 + p()->buff.windspeakers_lava_resurgence->data().effectN( 1 ).percent();
     }
 
+    if ( p()->opt_sl.t28_2pc_ele && p()->buff.fireheart->up() )
```

Comment:
```suggestion
    if ( p()->opt_sl.t28_2pc_ele && p()->buff.fireheart->check() )
```

Reply:
Why do you suggest to use `check`? If I understand it correctly `up` would provide the benefit tracking directly?

Reply:
Because the function can be called multiple times without actually executing the action, depending eg. on expressions.